src/Lecture4.hs

```diff
@@ -200,7 +244,25 @@ row in the file.
 -}
 
 rowToStats :: Row -> Stats
-rowToStats = error "TODO"
+rowToStats row = case rowTradeType row of
+   Buy -> template { statsTotalSum       = fmap (*(-1)) statsTotalSum template
```

Comment:
Using the `Functor` instance for `Sum` is something new! 😮 
But `Sum` has the `Num` instance so you can its methods to simplify the implementation a bit here:

```suggestion
   Buy -> template { statsTotalSum       = negate (statsTotalSum template)
```

Reply:
Yes, negate is simpler ! 
but why using fmap is new? I'm fmapping functions over some container since I started learning Haskell . I guess maybe it is something newcomers do more than necessary?

Reply:
I just generally haven't seen the usage of `Functor` data type for the `Sum` type 🙂 
Usually, people just wrap values using the `Sum` constructor, append then and extract. No other logic is usually used. But it doesn't mean that this data type must always be used like this 🙂